neo/Network/test_node_leader.py

```diff
@@ -231,8 +235,45 @@ def test_inventory_received(self):
 
         self.assertFalse(res2)
 
-        tx = self._generate_tx()
+        tx = self._generate_tx(Fixed8.TryParse(15))
 
         res = leader.InventoryReceived(tx)
 
         self.assertIsNone(res)
+
+    def _add_existing_tx(self):
+        wallet = self.GetWallet1()
+
+        res = []
+        for tx in wallet.GetTransactions():
+                json = tx.ToJson()
+                res.append(json)
+
+        txid = res[0]['txid']
+        tx, height = Blockchain.Default().GetTransaction(txid[2:])
+
+        tx_bytes = tx.Hash.ToBytes()
+        NodeLeader.Instance().MemPool[tx_bytes] = tx
+
+    def test_mempool_check_loop(self):
+        # delete any tx in the mempool
+        txs = []
+        keys = NodeLeader.Instance().MemPool.keys()
+        for i in keys:
+            tx = NodeLeader.Instance().GetTransaction(i)
+            txs.append(tx)
+
+        for tx in txs:
```

Comment:
same idea as previous comment. We can use `...MemPool.values()` instead of building a new list

Reply:
Ran into this error as discussed in Slack.
```
======================================================================
ERROR: test_mempool_check_loop (neo.Network.test_node_leader.LeaderTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/c/users/jseag/neo/neo-python-dev/neo/Network/test_node_leader.py", line 212, in test_mempool_check_loop
    NodeLeader.Instance().MempoolCheckLoop()
  File "/mnt/c/users/jseag/neo/neo-python-dev/neo/Network/NodeLeader.py", line 389, in MempoolCheckLoop
    for tx in self.MemPool.values():
RuntimeError: dictionary changed size during iteration

----------------------------------------------------------------------
```
Decided to build a new list from `.values()`